sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -566,12 +569,15 @@ case class DataSource(
   }
 
   private def disallowWritingIntervals(
-      dataTypes: Seq[DataType],
+      dataTypesWithNames: Seq[(DataType, String)],
+      format: String,
       forbidAnsiIntervals: Boolean): Unit = {
-    dataTypes.foreach(
-      TypeUtils.invokeOnceForInterval(_, forbidAnsiIntervals) {
-      throw QueryCompilationErrors.cannotSaveIntervalIntoExternalStorageError()
-    })
+    dataTypesWithNames.foreach { case (dataType, columnName) =>
+      TypeUtils.invokeOnceForInterval(dataType, forbidAnsiIntervals) {
+      throw QueryCompilationErrors.cannotSaveIntervalIntoExternalStorageError(
+        format, toSQLId(columnName), toSQLType(dataType)
```

Comment:
Better to follow existing convention, and apply quoting inside of `cannotSaveIntervalIntoExternalStorageError`

Reply:
Applied the comment. Thanks!